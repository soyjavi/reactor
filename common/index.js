import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

import Storage from './Storage';

export { LAYOUT, SHAPE, THEME, Storage };
