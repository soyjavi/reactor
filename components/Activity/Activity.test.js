import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import { Activity } from '..';

const { SPACE } = THEME;

const MOCKS = {
  COLOR: THEME.COLOR.ERROR,
  SIZE: 'large',
  STYLE: {
    padding: SPACE.M,
    backgroundColor: THEME.COLOR.BRAND,
  },
};

jest.useFakeTimers();

describe('<Activity>', () => {
  it('renders', () => {
    const tree = renderer.create(<Activity />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Activity color={MOCKS.COLOR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {size}', () => {
    const tree = renderer.create(<Activity size={MOCKS.SIZE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Activity style={MOCKS.STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
