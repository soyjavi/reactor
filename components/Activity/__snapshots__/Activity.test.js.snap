// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activity> renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
</View>
`;

exports[`<Activity> when {color} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#e85a7e",
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#e85a7e",
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#e85a7e",
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
</View>
`;

exports[`<Activity> when {size} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
</View>
`;

exports[`<Activity> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": undefined,
        "padding": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 8,
        "marginHorizontal": 4,
        "opacity": 0,
        "width": 8,
      }
    }
  />
</View>
`;
